wp/plugin/lib/cache.ml

```diff
@@ -74,33 +78,32 @@ module Project = struct
     Data.Cache.Service.request reader writer
 
   (* Loads project state (if any) from the cache. *)
-  let load digest : Project.state option =
+  let load (digest : digest) : Project.state option =
     let cache = project_cache () in
     Data.Cache.load cache digest
 
   (* Saves project state in the cache. *)
-  let save digest state : unit =
+  let save (digest : digest) (state : Project.state) : unit =
     let cache = project_cache () in
     Data.Cache.save cache digest state
 
 end
 
 module Program = struct
 
-  open KB.Let_syntax
-
   let package = "program"
 
   (* Gives a unique name to a KB object from the filename of the program. *)
-  let for_program filename =
+  let for_program (filename : string) : Theory.Unit.cls KB.obj KB.t =
```

Comment:
A package is a unit of administrative control, i.e., it is a namespace that is controlled by an organization or a sub-organization. You should take the package names that correspond to your group or your project, e.g., `draper`, `bap`, or `vibes`. When you need sub-organizations, sub-teams, or sub-projects, use sub-packages, e.g., `draper.vibes` or even `edu.cmu.ece.bap`, cf Java packages. CC @jtpaasch

Reply:
Thanks for letting us know! I'll update this right away.

Reply:
On the other hand, we also use packages like `file` in the knowledge base, but for objects, so here the package `program` fits nicely in the concept, but using the same package for the slot is definitely a no-no. Not only it tautological, i.e., you have a property `program:program` of the class `core-theory:program` but it also hinders composability as you program will not work if someone else will decide to add the same named property to the class program.

Reply:
Would the package `draper.cbat.wp` work? With this, we can have the property, `draper.cbat.wp:program`.

Reply:
it, will, but I believe that `draper` or `cbat` should be enough. I hope that you will be able to figure manage names in your ogranization.